### PR TITLE
Cleanup motion

### PR DESCRIPTION
**Pull request by Flova**
## Proposed changes
- Remove unused files and scripts
- Use ros2 numpy 
- Move head mover to motion
- Add typing to python code
- Restructure HCM dsd files
- Move state changes out of HCM descisions

# Related prs
https://github.com/bit-bots/bitbots_behavior/pull/334
https://github.com/bit-bots/bitbots_motion/pull/414
https://github.com/bit-bots/bitbots_navigation/pull/201
https://github.com/bit-bots/bitbots_lowlevel/pull/136
https://github.com/bit-bots/bitbots_misc/pull/237
https://github.com/bit-bots/udp_bridge/pull/26
https://github.com/bit-bots/humanoid_league_misc/pull/139
https://github.com/bit-bots/wolfgang_robot/pull/280
https://github.com/bit-bots/bitbots_msgs/pull/28

### CODE REVIEW

File: bitbots_head_mover/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+add_executable(move_head src/move_head.cpp)
+target_link_libraries(move_head
+  rclcpp::rclcpp
+  head_parameters
+)
+
+
+ament_target_dependencies(
+  move_head
+#head_capsule
```
Comment (jaagut) [inline bitbots_head_mover/CMakeLists.txt:44]: ???

File: bitbots_head_mover/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+moveit_ros_planning
+moveit_ros_planning_interface
+rclcpp
+sensor_msgs
+std_msgs
+tf2_geometry_msgs
+tf2_kdl
+tf2_ros
+
+
```
Comment (jaagut) [inline bitbots_head_mover/CMakeLists.txt:61]:
```suggestion
```

File: bitbots_head_mover/launch/head_mover_standalone.launch [hunk 3]
```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0"?>
+<launch>
+    <arg name="depends_only" default="false" />
```
Comment (jaagut) [inline bitbots_head_mover/launch/head_mover_standalone.launch:3]: What does this do? Please provide a description.

File: bitbots_head_mover/launch/head_mover_standalone.launch [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    <include file="$(find-pkg-share bitbots_utils)/launch/parameter_blackboard.launch">
+        <arg name="sim" value="$(var sim)"/>
+    </include>
+
+    <include file="$(find-pkg-share bitbots_robot_description)/launch/load_robot_description.launch" >
+        <arg name="sim" value="$(var sim)"/>
+    </include>
+
+    <include file="$(find-pkg-share bitbots_head_mover)/launch/head_mover.launch">
+        <arg name="depends_only" value="$(var depends_only)"/>
```
Comment (jaagut) [inline bitbots_head_mover/launch/head_mover_standalone.launch:15]: `head_mover.launch` does not contain this argument.

File: bitbots_head_mover/launch/head_mover_standalone.launch [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+        <param name="use_sim_time" value="$(var sim)" />
```
Comment (jaagut) [inline bitbots_head_mover/launch/head_mover_standalone.launch:16]: Instead, set the sim argument!

File: bitbots_head_mover/launch/test.launch [hunk 6]
```diff
@@ -0,0 +1,61 @@
+<launch>
+    <arg name="sim" default="false"/>
+    <arg name="viz" default="false"/>
+    <arg name="depends_only" default="false" />
```
Comment (jaagut) [inline bitbots_head_mover/launch/test.launch:4]: Description please.

File: bitbots_head_mover/launch/test.launch [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+        <arg name="sim" value="$(var sim)"/>
+    </include>
+
+    <group if="$(var viz)">
+        <!-- translate joint goals to joint states -->
+        <node pkg="bitbots_utils" exec="motor_goals_viz_helper.py" args="--head"/>
+        <!-- fake IMU needed for odometry-->
+        <node pkg="bitbots_utils" exec="dummy_imu.py"/>
+        <!-- add some visualization tools -->
+        <include file="$(find-pkg-share humanoid_league_interactive_marker)/launch/interactive_marker.launch" />
```
Comment (jaagut) [inline bitbots_head_mover/launch/test.launch:42]: Has been replaced by ros-sports

File: bitbots_head_mover/launch/test.launch [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+        <!-- create fake tf from map to robot -->
+        <node pkg="tf2_ros" exec="static_transform_publisher" name="static_map2odom_tf" args="--x -0.0 --y -0.0 --z 0.0 --qx 0.0 --qy 0.0 --qz 0.0 --qw 1.0 --frame-id map --child-frame-id odom" />
+    </group>
+
+    <group unless="$(var viz)">
+        <include file="$(find-pkg-share bitbots_vision)/launch/vision_startup.launch" >
+            <arg name="sim" value="$(var sim)"/>
+        </include>
+        <include file="$(find-pkg-share soccer_ipm)/launch/transformer.launch" >
```
Comment (jaagut) [inline bitbots_head_mover/launch/test.launch:51]: This file does not exist.

File: bitbots_head_mover/package.xml [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+      point on the field, e.g. to track the ball.</description>
+  <maintainer email="valerie.bartel@uni-hamburg.de">Valerie Bartel</maintainer>
+  <maintainer email="info@bit-bots.de">Hamburg Bit-Bots</maintainer>
+  <author email="info@bit-bots.de">Hamburg Bit-Bots</author>
+
+  <license>MIT</license>
+
+  <buildtool_depend>ament_cmake</buildtool_depend>
+
+  <depend>backward_ros</depend>
```
Comment (jaagut) [inline bitbots_head_mover/package.xml:18]: Should also depend on the parameter blackboard

**Review by jaagut — COMMENTED** [review]
First review of ~3/5 of the more easy files…

File: bitbots_hcm/bitbots_hcm/hcm_dsd/actions/wait_for.py [hunk 10]
```diff
@@ -1,56 +1,31 @@
...
 
-from dynamic_stack_decider.abstract_action_element import AbstractActionElement
 
-
-class WaitForIMUStartup(AbstractActionElement):
+class WaitForIMUStartup(AbstractHCMActionElement):
     """
     Waits for the IMU to connect and does not complain as we are still in start up.
     """
+    pass
```
[the remark this answers is not in the document]
Comment (Flova) [reply bitbots_hcm/bitbots_hcm/hcm_dsd/actions/wait_for.py:8]: They are essentially a print / wait with a type for better understanding

[the remark this answers is not in the document]
Comment (Flova) [reply bitbots_hcm/bitbots_hcm/hcm_dsd/actions/wait_for.py:8]: The check is done in the decision

File: bitbots_hcm/config/hcm_wolfgang.yaml [hunk 11]
```diff
@@ -1,62 +1,41 @@
...
     # Motors
     motor_off_time: 30000000.0 # time of no use or updates when the hcm goes to soft off
     motor_timeout_duration: 0.5 # time without messages from the servos till error is produced [s]
 
-    imu_timeout_duration: 0.5 # time without messages from the IMU till error is produced [s]
-
-    #Animations
-    anim_server_wait_time: 1.0 # time the hcm waits for animation server
+    # Animations
+    anim_server_wait_time: 10.0 # time the hcm waits for animation server
```
[the remark this answers is not in the document]
Comment (Flova) [reply bitbots_hcm/config/hcm_wolfgang.yaml:14]: What do you mean? I think this is the wait at startup.

File: bitbots_hcm/package.xml [hunk 12]
```diff
@@ -20,16 +20,18 @@
...
+  <depend>bitbots_robot_description</depend>
+  <depend>bitbots_utils</depend>
   <depend>geometry_msgs</depend>
-  <depend>humanoid_league_msgs</depend>
   <depend>pybind11-dev</depend>
+  <depend>python3-numpy</depend>
+  <depend>python3-sklearn</depend>
+  <depend>python3-transforms3d</depend>
   <depend>rclpy</depend>
+  <depend>ros2_numpy</depend>
```
[the remark this answers is not in the document]
Comment (Flova) [reply bitbots_hcm/package.xml:30]: It is. It interfaces between numpy and messages.

File: bitbots_head_mover/package.xml (hunk 9 above)
Comment (Flova) [reply bitbots_head_mover/package.xml:18]: This is a part of bitbots_utils which is already included

File: bitbots_hcm/bitbots_hcm/hcm_dsd/hcm_blackboard.py [hunk 13]
```diff
@@ -4,142 +4,112 @@
...
-        self.start_time = self.current_time
-        # Imu
+
+        # Save start time
+        self.start_time: Time = self.node.get_clock().now()
+
+        # Get parameters
+        self.simulation_active: bool = self.node.get_parameter('simulation_active').value
+        self.visualization_active: bool = self.node.get_parameter('visualization_active').value
+        self.pickup_accel_threshold: float = 1000.0  # TODO: make this a parameter
```
Comment (jaagut) [inline bitbots_hcm/bitbots_hcm/hcm_dsd/hcm_blackboard.py:35]: This is already a param, at least it exists in the config file

File: bitbots_hcm/bitbots_hcm/hcm_dsd/hcm_blackboard.py [hunk 14]
```diff
@@ -4,142 +4,112 @@
...
-        self.motor_off_animation = self.node.get_parameter("animations.motor_off").value
-        self.stand_up_front_animation = self.node.get_parameter("animations.stand_up_front").value
-        self.stand_up_back_animation = self.node.get_parameter("animations.stand_up_back").value
-        self.stand_up_left_animation = self.node.get_parameter("animations.stand_up_left").value
-        self.stand_up_right_animation = self.node.get_parameter("animations.stand_up_right").value
-        # motors
-        # initialize with current time, or motors will be turned off on start
-        self.last_motor_goal_time = self.node.get_clock().now()
+        self.external_animation_running: bool = False
+        self.last_animation_goal_time: Time = self.node.get_clock().now()
```
Comment (jaagut) [inline bitbots_hcm/bitbots_hcm/hcm_dsd/hcm_blackboard.py:69]: For consistency, should be None

File: bitbots_hcm/bitbots_hcm/hcm_dsd/hcm_blackboard.py [hunk 15]
```diff
@@ -4,142 +4,112 @@
...
+        self.animation_name_falling_front: str = self.node.get_parameter("animations.falling_front").value
+        self.animation_name_falling_back: str = self.node.get_parameter("animations.falling_back").value
+        self.animation_name_falling_left: str = self.node.get_parameter("animations.falling_left").value
+        self.animation_name_falling_right: str = self.node.get_parameter("animations.falling_right").value
+        self.animation_name_turning_back_left: str = self.node.get_parameter("animations.turning_back_left").value
+        self.animation_name_turning_back_right: str = self.node.get_parameter("animations.turning_back_right").value
+
+        # Motor State
+        # Initialize with current time, or motors will be turned off on start
+        self.last_motor_goal_time: Time = self.node.get_clock().now()
```
Comment (jaagut) [inline bitbots_hcm/bitbots_hcm/hcm_dsd/hcm_blackboard.py:82]: For consistency, should be None

File: bitbots_hcm/bitbots_hcm/hcm_dsd/hcm_blackboard.py [hunk 16]
```diff
@@ -4,142 +4,112 @@
...
         self.last_motor_update_time =  Time()
```
Comment (jaagut) [inline bitbots_hcm/bitbots_hcm/hcm_dsd/hcm_blackboard.py:83]: For consistency, should be None

File: bitbots_hcm/bitbots_hcm/hcm_dsd/hcm_blackboard.py [hunk 17]
```diff
@@ -4,142 +4,112 @@
...
         self.current_joint_state: Optional[JointState] = None
         self.previous_joint_state: Optional[JointState] = None
-        anim_package = self.node.get_parameter("animations.anim_package").value
-        path = get_package_share_directory(anim_package)
-        path = os.path.join(path, 'animations/motion/', self.walkready_animation + '.json')
-        with open(path, 'r') as f:
-            json_data = json.load(f)
-        keyframes: List[dict] = json_data["keyframes"]
-        self.walkready_pose_dict: dict = keyframes[-1]["goals"]
-        self.walkready_pose_threshold = self.node.get_parameter("animations.walkready_pose_threshold").value
```
Comment (jaagut) [inline bitbots_hcm/bitbots_hcm/hcm_dsd/hcm_blackboard.py:95]: I hope, this is now handled by the dynup or animation server?

File: bitbots_hcm/bitbots_hcm/humanoid_control_module.py [hunk 18]
```diff
@@ -1,203 +1,195 @@
...
-from builtin_interfaces.msg import Time as TimeMsg
 
-class HardwareControlManager:
 
+class HardwareControlManager:
     def __init__(self, use_sim_time, simulation_active, visualization_active):
         rclpy.init(args=None)
         node_name = "hcm_py"
+
+        self.simulation_active = simulation_active
```
Comment (jaagut) [inline bitbots_hcm/bitbots_hcm/humanoid_control_module.py:34]:
```suggestion
```
This is never used

File: bitbots_hcm/bitbots_hcm/humanoid_control_module.py [hunk 19]
```diff
@@ -1,203 +1,195 @@
...
-        self.blackboard.shut_down_request = True
-        self.node.get_logger().warn("You're stopping the HCM. The robot will sit down and power off its motors.")
-        speak("Stopping HCM", self.blackboard.speak_publisher, priority=50)
-        # now wait for it finishing the shutdown procedure
-        while not self.blackboard.current_state == RobotControlState.HCM_OFF:
-            # we still have to update everything
-            self.blackboard.current_time = self.node.get_clock().now()
-            self.dsd.update()
-            self.hcm_state_publisher.publish(self.blackboard.current_state)
-            self.node.get_clock().sleep_for(Duration(seconds=0.01))
```
Comment (jaagut) [inline bitbots_hcm/bitbots_hcm/humanoid_control_module.py:143]: Is this handled somewhere else or removed entirely?

**Review by jaagut — COMMENTED** [review]
🎉 Finally, finished all files!

File: bitbots_animation_server/bitbots_animation_server/animation_node.py [hunk 20]
```diff
@@ -135,65 +136,52 @@ def execute_cb(self, goal):
...
+            traj_msg.points = [JointTrajectoryPoint()]
             # We are only using a single point in the trajectory message, since we only want to send a single joint goal
-            self.traj_msg.points[0].positions = []
-            self.traj_msg.points[0].effort = []
+            traj_msg.points[0].positions = []
+            traj_msg.points[0].effort = []
             for joint in pose:
-                self.traj_msg.joint_names.append(joint)
-                self.traj_msg.points[0].positions.append(pose[joint])
+                traj_msg.joint_names.append(joint)
```
[the remark this answers is not in the document]
Comment (Flova) [reply bitbots_animation_server/bitbots_animation_server/animation_node.py:177]: I don't know what you mean.

File: bitbots_animation_server/bitbots_animation_server/resource_manager.py [hunk 21]
```diff
@@ -176,26 +154,3 @@ def find_all_animations_by_name(self, force_reload=False):
...
-            self.find_all_animations(force_reload=True)
-        return name in self.names
-
-_RM = None  # type: ResourceManager
-# Shortcut to search for animations
-def find_all_animations_by_name(node, *args, **kwargs):
-    global _RM
-    if not _RM:
-        _RM = ResourceManager(node)
-    return _RM.find_all_animations_by_name(*args, **kwargs)
```
[the remark this answers is not in the document]
Comment (Flova) [reply bitbots_animation_server/bitbots_animation_server/resource_manager.py:201]: Exactly

File: bitbots_hcm/src/hcm.cpp [hunk 22]
```diff
@@ -210,50 +225,68 @@ class HCM_CPP : public rclcpp::Node {
...
   bitbots_msgs::msg::FootPressure current_pressure_right_;
-  geometry_msgs::msg::PointStamped current_cop_left_;
-  geometry_msgs::msg::PointStamped current_cop_right_;
   sensor_msgs::msg::JointState current_joint_state_;
+
+  // Walking state
   builtin_interfaces::msg::Time last_walking_time_;
+
+  // Kick state
+  builtin_interfaces::msg::Time last_kick_time_;
```
[the remark this answers is not in the document]
Comment (Flova) [reply bitbots_hcm/src/hcm.cpp:276]: They are used to determine if we recently got data from the walking or kick. Dynup is not needed, as the HCM is the only one calling it, so we know when it is active that way.

File: bitbots_hcm/bitbots_hcm/hcm_dsd/decisions/fallen.py [hunk 23]
```diff
@@ -0,0 +1,56 @@
...
+    """
+    def __init__(self, blackboard, dsd, parameters=None):
+        super().__init__(blackboard, dsd, parameters)
+
+        # Get parameters
+        self.fallen_orientation_thresh = math.radians(self.blackboard.node.get_parameter("fallen_orientation_thresh").value)
+        self.fallen_angular_velocity_thresh = self.blackboard.node.get_parameter("fallen_angular_velocity_thresh").value
+
+    def perform(self, reevaluate=False):
+        # Check of the fallen detection is active
```
[the remark this answers is not in the document]
Comment (Flova) [reply bitbots_hcm/bitbots_hcm/hcm_dsd/decisions/fallen.py:22]: This is not the case here right?

[the remark this answers is not in the document]
Comment (Flova) [reply bitbots_hcm/bitbots_hcm/hcm_dsd/decisions/fallen.py:22]: The check checks if we generally want to perform any standup motions. If not no Fall detection is performed.

[the remark this answers is not in the document]
Comment (Flova) [reply bitbots_hcm/bitbots_hcm/hcm_dsd/decisions/fallen.py:22]: This is unrelated to the current dynup state

File: bitbots_hcm/bitbots_hcm/hcm_dsd/decisions/kicking.py [hunk 24]
```diff
@@ -0,0 +1,30 @@
...
+            return "NOT_KICKING"
+
+        # Calculate the time delta between now and the last kick goal
+        time_delta = self.blackboard.node.get_clock().now().nanoseconds / 1e9 - self.blackboard.last_kick_goal_time.nanoseconds / 1e9
+
+        # Log the time delta between now and the last kick goal
+        self.publish_debug_data("Last Kick Goal Time Delta", time_delta)
+
+        # If the time delta is smaller enough, we are still kicking
+        if time_delta < 0.1:
```
[the remark this answers is not in the document]
Comment (Flova) [reply bitbots_hcm/bitbots_hcm/hcm_dsd/decisions/kicking.py:23]: It was that way and while ugly it works. I would prefer to not change that logic.

File: bitbots_hcm/bitbots_hcm/hcm_dsd/decisions/kicking.py [hunk 25]
```diff
@@ -0,0 +1,30 @@
...
+            # we are walking and can stay like this
```
[the remark this answers is not in the document]
Comment (Flova) [reply bitbots_hcm/bitbots_hcm/hcm_dsd/decisions/kicking.py:24]: Probably a copy and paste error. These lines are not new, only moved because we split the decision file. We can still fix this tho..

File: bitbots_hcm/bitbots_hcm/hcm_dsd/decisions/pickup.py [hunk 26]
```diff
@@ -0,0 +1,30 @@
+from bitbots_hcm.hcm_dsd.decisions import AbstractHCMDecisionElement
+from humanoid_league_speaker.speaker import speak
+
+from bitbots_msgs.msg import RobotControlState
+
+
+class PickedUp(AbstractHCMDecisionElement):
+    """
+    Decides if the robot is currently picked up
```
[the remark this answers is not in the document]
Comment (Flova) [reply bitbots_hcm/bitbots_hcm/hcm_dsd/decisions/pickup.py:9]: I don't think the original one is wrong tho..

File: bitbots_hcm/bitbots_hcm/hcm_dsd/hcm_blackboard.py (hunk 14 above)
Comment (Flova) [reply bitbots_hcm/bitbots_hcm/hcm_dsd/hcm_blackboard.py:69]: I agree

File: bitbots_hcm/bitbots_hcm/hcm_dsd/hcm_blackboard.py (hunk 17 above)
Comment (Flova) [reply bitbots_hcm/bitbots_hcm/hcm_dsd/hcm_blackboard.py:95]: The dynup is not fully responsible for the walkready pose.

File: bitbots_hcm/bitbots_hcm/humanoid_control_module.py (hunk 19 above)
Comment (Flova) [reply bitbots_hcm/bitbots_hcm/humanoid_control_module.py:143]: It is not needed anymore, as we are unable to perform a sit down motion.

File: bitbots_hcm/bitbots_hcm/hcm_dsd/hcm_blackboard.py (hunk 17 above)
Comment (jaagut) [reply bitbots_hcm/bitbots_hcm/hcm_dsd/hcm_blackboard.py:95]: I was just asking because this code part was removed

File: bitbots_hcm/bitbots_hcm/hcm_dsd/decisions/fallen.py (hunk 23 above)
[the remark this answers is not in the document]
Comment (jaagut) [reply bitbots_hcm/bitbots_hcm/hcm_dsd/decisions/fallen.py:22]: Okay, but still the original comment is not good

File: bitbots_hcm/bitbots_hcm/hcm_dsd/hcm_blackboard.py (hunk 17 above)
Comment (Flova) [reply bitbots_hcm/bitbots_hcm/hcm_dsd/hcm_blackboard.py:95]: This was a autocomplete error. I meant to say
"The dynup is **now** fully responsible for the walkready pose."

Comment (Flova) [reply bitbots_hcm/bitbots_hcm/hcm_dsd/hcm_blackboard.py:95]: Therefore this can be removed

File: bitbots_hcm/src/hcm.cpp [hunk 27]
```diff
@@ -149,26 +166,31 @@ class HCM_CPP : public rclcpp::Node {
...
 
   void dynup_callback(const bitbots_msgs::msg::JointCommand msg) {
-    if (current_state_ == humanoid_league_msgs::msg::RobotControlState::STARTUP ||
-        current_state_ == humanoid_league_msgs::msg::RobotControlState::FALLEN ||
-        current_state_ == humanoid_league_msgs::msg::RobotControlState::GETTING_UP ||
-        current_state_ == humanoid_league_msgs::msg::RobotControlState::CONTROLLABLE) {
+    if (current_state_ == bitbots_msgs::msg::RobotControlState::STARTUP ||
+        current_state_ == bitbots_msgs::msg::RobotControlState::GETTING_UP ||
+        current_state_ == bitbots_msgs::msg::RobotControlState::MOTOR_OFF ||
+        current_state_ == bitbots_msgs::msg::RobotControlState::PICKED_UP ||
```
[the remark this answers is not in the document]
Comment (Flova) [reply bitbots_hcm/src/hcm.cpp:172]: Because we might want to play the sitdown down in case of an overload or play the walkready animation if we are picked up so we can be placed down nicely afterwards. The later is arguably more important, as we don't really perform any sit down motions due to the springs.

File: bitbots_hcm/bitbots_hcm/hcm_dsd/actions/play_animation.py [hunk 28]
```diff
@@ -1,21 +1,17 @@
...
-import rclpy
-from bitbots_msgs.action import Dynup
-from dynamic_stack_decider.abstract_action_element import AbstractActionElement
-from humanoid_league_msgs.action import PlayAnimation
-from humanoid_league_msgs.msg import RobotControlState
+from bitbots_msgs.action import Dynup, PlayAnimation
 
 
-class AbstractPlayAnimation(AbstractActionElement):
+class AbstractPlayAnimation(AbstractHCMActionElement):
```
[the remark this answers is not in the document]
Comment (Flova) [reply bitbots_hcm/bitbots_hcm/hcm_dsd/actions/play_animation.py:8]: Done

File: bitbots_hcm/bitbots_hcm/hcm_dsd/decisions/animation.py [hunk 29]
```diff
@@ -0,0 +1,32 @@
+from bitbots_hcm.hcm_dsd.decisions import AbstractHCMDecisionElement
+
+
+class Record(AbstractHCMDecisionElement):
```
[the remark this answers is not in the document]
Comment (Flova) [reply bitbots_hcm/bitbots_hcm/hcm_dsd/decisions/animation.py]: Done

File: bitbots_hcm/bitbots_hcm/hcm_dsd/hcm_blackboard.py (hunk 13 above)
Comment (Flova) [reply bitbots_hcm/bitbots_hcm/hcm_dsd/hcm_blackboard.py]: Done

File: bitbots_hcm/bitbots_hcm/hcm_dsd/hcm_blackboard.py (hunk 14 above)
Comment (Flova) [reply bitbots_hcm/bitbots_hcm/hcm_dsd/hcm_blackboard.py:65]: Reworked this whole thing

File: bitbots_hcm/bitbots_hcm/hcm_dsd/hcm_blackboard.py (hunk 15 above)
Comment (Flova) [reply bitbots_hcm/bitbots_hcm/hcm_dsd/hcm_blackboard.py]: Same here

File: bitbots_hcm/bitbots_hcm/hcm_dsd/hcm_blackboard.py (hunk 16 above)
Comment (Flova) [reply bitbots_hcm/bitbots_hcm/hcm_dsd/hcm_blackboard.py]: Same here

File: bitbots_hcm/bitbots_hcm/humanoid_control_module.py (hunk 18 above)
Comment (Flova) [reply bitbots_hcm/bitbots_hcm/humanoid_control_module.py]: You are correct. I removed it

File: bitbots_hcm/bitbots_hcm/hcm_dsd/decisions/check_hardware.py [hunk 30]
```diff
@@ -13,50 +11,44 @@ class CheckMotors(AbstractHCMDecisionElement):
...
+        if self.blackboard.visualization_active:
+            # we will have no problems with hardware in visualization
+            return "OKAY"
 
         if self.blackboard.simulation_active:
-            # Some simulators will give exact same joint messages which look like errors, so ignore this case
-            if self.blackboard.last_motor_update_time != Time(seconds=int(0), nanoseconds=0 % 1 * 1e9):
-                return "OKAY"
-            else:
+            # Some simulators will give exact same joint messages could look like errors,
```
Comment (jaagut) [inline bitbots_hcm/bitbots_hcm/hcm_dsd/decisions/check_hardware.py:33]:
```suggestion
            # Some simulators will give the exact same joint messages, which could look like errors,
```

File: bitbots_hcm/bitbots_hcm/hcm_dsd/hcm.dsd [hunk 31]
```diff
@@ -1,48 +1,37 @@
...
+        FREE -->$RecordAnimation
+            RECORD_ACTIVE --> @RobotStateRecord, @Wait
             FREE --> $CheckMotors
-                MOTORS_NOT_STARTED --> @WaitForMotorStartup
-                OVERLOAD --> @PlayAnimationDynup + direction:descend, @TurnMotorsOff, @StayMotorsOff
-                PROBLEM --> @WaitForMotors
-                TURN_OFF --> @PlayAnimationDynup + direction:descend, @TurnMotorsOff, @StayMotorsOff
-                TURN_ON --> @TurnMotorsOn, @PlayAnimationDynup + direction:rise
+                MOTORS_NOT_STARTED --> @RobotStateStartup, @WaitForMotorStartup
+                OVERLOAD --> @RobotStateMotorOff, @CancelGoals, @StopWalking, @PlayAnimationFallingFront, @TurnMotorsOff, @Wait
```
Comment (jaagut) [inline bitbots_hcm/bitbots_hcm/hcm_dsd/hcm.dsd:10]: Why did you change it to play falling front animation when we are having overload errors?

**Review by jaagut — APPROVED** [review]

File: bitbots_animation_server/bitbots_animation_server/animation_node.py (hunk 20 above)
[the remark this answers is not in the document]
Comment (jaagut) [reply bitbots_animation_server/bitbots_animation_server/animation_node.py:177]: I don't remember the solution approach. But the thing I was referring to is the (old) quirk of the HCM flag in the PlayAnimation call and the animation server handling it different depending on the flag.

File: bitbots_hcm/bitbots_hcm/hcm_dsd/hcm.dsd (hunk 31 above)
Comment (Flova) [reply bitbots_hcm/bitbots_hcm/hcm_dsd/hcm.dsd:10]: Because sitting down might not be desirable if we have an overload and springs in the knees. The falling front animation will result in the robot falling to the front in a more or less controlled manner, which is probably better.

File: bitbots_animation_server/bitbots_animation_server/animation_node.py (hunk 20 above)
[the remark this answers is not in the document]
Comment (Flova) [reply bitbots_animation_server/bitbots_animation_server/animation_node.py:177]: I don't remember this. Be we talked about having the head safe states as part of the head mover, so it could select an optimal orientation.

[the remark this answers is not in the document]
Comment (jaagut) [reply bitbots_animation_server/bitbots_animation_server/animation_node.py:177]: If not yet, please open an issue.
